src/Capgemini.PowerApps.PackageDeployerTemplate/Services/FlowActivationService.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Capgemini.PowerApps.PackageDeployerTemplate.Adapters;
+using Microsoft.Extensions.Logging;
+using Microsoft.Xrm.Sdk;
+using Microsoft.Xrm.Sdk.Query;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Capgemini.PowerApps.PackageDeployerTemplate.Services
+{
+    public class FlowActivationService
+    {
+        private readonly ILogger logger;
+        private readonly ICrmServiceAdapter crmSvc;
+
+        public FlowActivationService(ILogger logger, ICrmServiceAdapter crmSvc)
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this.crmSvc = crmSvc ?? throw new ArgumentNullException(nameof(crmSvc));
+        }
+
+        public void ActivateFlows(IEnumerable<string> flowsToIgnore, IEnumerable<string> solutions)
+        {         
+            if (solutions == null || !solutions.Any())
+            {
+                logger.LogInformation($"No solutions to activate flows for");
+                return;
+            }
+
+            foreach (var item in solutions)
+            {
+                var solutionId = GetSolutionIdByUniqueName(item);
+                if (!solutionId.HasValue) return;
+
+                var solutionWorkflowIds = this.GetSolutionComponentObjectIdsByType(solutionId.Value, 29);
+                if (!solutionWorkflowIds.Any()) return;
+
+                var solutionFlowIds = this.GetDeployedFlows(solutionWorkflowIds, new ColumnSet("name"));
+
+                logger.LogInformation($"Flows found {solutionFlowIds.Count()}");
+
+                foreach (var solutionFlow in solutionFlowIds)
+                {
+                    string matchedFlow = null;
+
+                    if (flowsToIgnore !=null)
+                    {
+                         matchedFlow = flowsToIgnore.FirstOrDefault(f => f == solutionFlow.Attributes["name"].ToString());
+                    }                   
+                    
+                    var stateCode =  matchedFlow == null ? 1 : 0;
```

Comment:
Could these magic number also have names to avoid confusion?

Reply:
Yep constants.cs will sort this.